src/tooling/tooling-editor-plugin.md

```diff
@@ -0,0 +1,57 @@
+<!--
+  ~ Copyright (c) godot-rust; Bromeon and contributors.
+  ~ This Source Code Form is subject to the terms of the Mozilla Public
+  ~ License, v. 2.0. If a copy of the MPL was not distributed with this
+  ~ file, You can obtain one at https://mozilla.org/MPL/2.0/.
+-->
+
+# Tooling with Editor Plugin
+
+Using `EditorPlugin` types is very similar to the process used when
+[writing plugins in GDScript](https://docs.godotengine.org/en/stable/tutorials/plugins/editor/making_plugins.html). However, like Simple Types, they
+are registered automatically by godot-rust and cannot be enabled/disabled in the Project Settings plugins pane.
+
+Typically, Plugins are automatically disabled if they have a code error, but because Rust is a compiled language, you (typically) cannot update the
+code without removing any errors.
+
+
+## Creating an EditorPlugin
+
+```rs
+
+#[derive(GodotClass)]
+#[class(tool, editor_plugin, base=EditorPlugin)]
+struct MyEditorPlugin {
+    #[base]
+    base : Base<EditorPlugin>
+}
+
+impl IEditorPlugin for MyEditorPlugin {
+    fn init(base : Base<EditorPlugin>) -> Self {
+        // this is a prime candidate to be replaced by "init"
+        Self {
+            base
+        }
+    }
+
+    fn enter_tree(&mut self) {
+        // perform typical plugin operations here
+
+    }
+
+    fn exit_tree(&mut self) {
+        // perform typical plugin operations here
+
+    }
+}
+```
+
+```admonish hint
+Since this is a plugin, that custom node of yours will be added to the scene tree and visible during runtime. This means you can access the scene 
+tree with this. Additionally, it is safe to access the `EditorInterface` singleton through this node, which allows adding different GUI elements to 
+the editor directly. This can be incredibly helpful if you have an advanced GUI you want to implement.
+
+Use an `is_editor_hint` [guard](https://en.wikipedia.org/wiki/Guard_(computer_science)) if you don't want some code executing 
+during runtime of the game.
```

Comment:
I can do that. I was taking the opportunity to introduce some people to guard clauses since those are super helpful especially when working with `Option<T>` and `Result<T, _>`

Reply:
It's also fine if you link to both 🙂